core/framework/storage.py

```
@@ -0,0 +1,161 @@
+    """
+    Manage database connections and provide storage utilities.
+    
+    Features:
+    - Connection pooling
+    - Automatic retry logic
+    - Collection management
+    - Index management
+    """
+    
+    # Cache connections to avoid recreating
+    _connections: Dict[str, StandardDatabase] = {}
```

Ruff (0.12.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)